### DISCUSSION TITLE
Ongoing list of Data Validation & Default Settings

### DISCUSSION THREAD

**Discussion by abelsiqueira** [TulipaEnergy/TulipaEnergyModel.jl#396] [Ideas]
For model data structures in [src/structures.jl](../blob/main/src/structures.jl) like `GraphAssetData`, as we use more complex data, we should check for consistency. E.g. if we make `GraphAssetData.investable` an array of years, `investment_cost` should also be an array of equal length.

**Comment by abelsiqueira** [comment]
This will only make sense after multi-year investment is implemented (maybe we need an issue for that?) because otherwise there are no vectors in the input data, except for profiles and partitions.
So we can move this to M2.

**Comment by clizbe** [comment]
Hmm you're sure it won't make sense till multi-year? By the schedule we won't do multi-year until M3... 
Or is this something we should do a basic version now and improve it?

**Comment by abelsiqueira** [comment]
I don't see how it can be done right now, because there is nothing to be checked. A single `GraphAssetData` stores data from a single asset. On creation, we use only input data (no profiles, no partition), so there are no vectors. Maybe @suvayu  has a different view.
As soon as we introduce multi-year data, this can, and should, be implemented.

**Comment by suvayu** [comment]
I used the multi-year as an off-the-cuff example.  There could be others.  Maybe we should create an epic issue for checks like this, and collect all ideas there.  We can start working on them once the ideas start becoming important.

I'm primarily worried about forgetting about details like this.  So if you think an issue is not the right place to record them, I'm fine recording them elsewhere as long as nothing slips through the cracks.

**Comment by datejada** [comment]
Consistency checks for connections of the assets: We shouldn't allow a transport asset to be the charging flow for a storage asset (but double-check).

**Comment by datejada** [comment]
We need to add the following validations:

- Check that all assets have a type associated (i.e., consumer, producer, conversion, storage, hub)
- Check that the flows that are defined as investable are also defined as transport

**Comment by clizbe** [comment]
Has been moved to #461
